core/src/main/java/org/apache/carbondata/core/datastore/page/encoding/adaptive/AdaptiveDeltaIntegralCodec.java

```diff
@@ -88,7 +97,32 @@ public ColumnPageEncoder createEncoder(Map<String, String> parameter) {
         encodedPage = ColumnPage.newPage(input.getColumnSpec(), targetDataType,
             input.getPageSize());
         input.convertValue(converter);
-        byte[] result = encodedPage.compress(compressor);
+        result = encodedPage.compress(compressor);
+        if (isInvertedIndex) {
```

Comment:
can u please move this code to some utility and as below classes also has same logic